src/genworld.cpp

```diff
@@ -337,3 +341,147 @@ void GenerateWorld(GenWorldMode mode, uint size_x, uint size_y, bool reset_setti
 
 	_GenerateWorld();
 }
+
+/** Town data imported from JSON files and used to place towns. */
+struct ExternalTownData {
+	std::string name;   ///< The name of the town.
+	uint population;    ///< The target population of the town when created in OpenTTD. If input is blank, defaults to 0.
+	bool is_city;       ///< Should it be created as a city in OpenTTD? If input is blank, defaults to false.
+	float x_proportion; ///< The X coordinate of the town, as a proportion 0..1 of the maximum X coordinate.
+	float y_proportion; ///< The Y coordinate of the town, as a proportion 0..1 of the maximum Y coordinate.
+};
+
+/**
+ * Helper for CircularTileSearch to found a town on or near a given tile.
+ * @param tile The tile to try founding the town upon.
+ * @param user_data The ExternalTownData to attempt to found.
+ */
+static bool TryFoundTownNearby(TileIndex tile, void *user_data)
+{
+	ExternalTownData town = *(ExternalTownData *)user_data;
```

Comment:
*static_cast<ExternalTownData *>(user_data);

Reply:
Presumably this should also be `const ExternalTownData &town =`?

Reply:
Good point, it's making a copy of it right now, whoops!